backend/src/modules/match-module/core/domain/Match.ts

```diff
@@ -0,0 +1,63 @@
+import { MatchHasStarted } from './event/MatchHasStarted';
+import { MatchId } from './MatchId';
+import { MatchSideId } from './MatchSideId';
+import { DomainCommandResult } from '../../../../shared/core/domain/DomainCommandResult';
+
+export class Match {
+  readonly matchId: MatchId;
+  readonly firstTeamId: MatchSideId;
+  readonly secondTeamId: MatchSideId;
+  readonly winner: MatchSideId | undefined;
+  readonly looser: MatchSideId | undefined;
+  readonly hasEnded: boolean | undefined = false;
+
+  constructor(props: {
+    matchId: MatchId;
+    firstTeamId: MatchSideId;
+    secondTeamId: MatchSideId;
+    winner?: MatchSideId;
+    looser?: MatchSideId;
+    hasEnded?: boolean;
+  }) {
+    this.matchId = props.matchId;
+    this.firstTeamId = props.firstTeamId;
+    this.secondTeamId = props.secondTeamId;
+    this.winner = props.winner;
+    this.looser = props.looser;
+    this.hasEnded = props.hasEnded;
+  }
+}
+
+export function startMatch(
+  state: Match | undefined,
+  command: { matchId: string; firstTeamId: string; secondTeamId: string },
+  currentTime: Date,
+): DomainCommandResult<Match> {
+  if (state?.matchId !== undefined) {
+    throw new Error('Cannot start a match that has already begun.');
+  }
+  if (!command.firstTeamId || !command.secondTeamId) {
```

Comment:
Może drużyna grać sama że sobą :)?

Reply:
ach głupi ja :D

Reply:
to jeszcze pytanie - skoro tutaj są już typy domenowe to chyba sprawdzanie powyższego warunku nie ma sensu, bo aplikacja wcześniej się wysypie jak do command handlera wpadnie pusty string, no nie? tutaj zadziała nam walidacja z value object MatchSideId

Reply:
Tak, to prawda, po to są też te typy :) Ale wtedy to co napisałem, czyli, że nie może grać sama ze soba dalej jest istotne. 
To bym zrealizował jakąś metodą w MatchSideId np. 
`equals(other: MatchSideId | undefined)`
Żeby enkapsulować to w klasie a nie wyjmować z obu raw i potem robić porównania, co może być już w samej klasie. W sumie domena nigdy nie powinna się dobierać do wartości raw (poza mapowaniem na event - bo w naszym przypadku eventu używamy w 2 zastosowaniach - w architekturze bardziej rozbudowanej event miałby jeszcze jedną warstwę i byłby DomainEvent i ApplicationEvent - tutaj już i tak jest dużo i nie chciałem tego wproawdzać).

Reply:
okej czyli w tym warunku sprawdzam czy firstTeamId i secondTeamId są tego samego typu, jeśli tak to wszystko git lecimy dalej, jeśli nie to wyrzucam błąd, że nie ma jednej z drużyn (?)